src/types/capabilities.rs

```diff
@@ -31,21 +33,28 @@ use std::hash::Hash;
 pub struct Capabilities(
     // Note that this field isn't *actually* 'static.
     // Rather, it is tied to the lifetime of the `ZeroCopy` that contains this `Name`.
-    pub(crate) HashSet<&'static str>,
+    pub(crate) HashSet<Capability<'static>>,
 );
 
 impl Capabilities {
     /// Check if the server has the given capability.
-    pub fn has<S: ?Sized>(&self, s: &S) -> bool
-    where
-        for<'a> &'a str: Borrow<S>,
-        S: Hash + Eq,
-    {
+    pub fn has<'a>(&self, s: &Capability<'a>) -> bool {
         self.0.contains(s)
     }
 
+    /// Check if the server has the given capability via str.
+    pub fn has_str(&self, s: &str) -> bool {
```

Comment:
I'd prefer for this to be generic over the type of string the same way the old one was.